lib/user_comm_screener.rb

```diff
@@ -26,6 +26,11 @@
 #
 # An important note is that **all of these settings do not apply when the actor
 # is a staff member**. So admins and moderators can PM and notify anyone they please.
+#
+# The secondary usage of this class is to determine which users the actor themselves
```

Comment:
Given that this is not even used in core, can the plugin which relies on this method define an custom class for the logic that needs to be added? Looking at the implementation of the methods, the only reliance is on `@target_users` which can be passed as an argument to the custom class.

Reply:
I don't see why we would put this in a plugin rather than in the core class? This is useful for more than what the chat plugin uses it for. Would much prefer to keep it here in one place.

Reply:
> I don't see why we would put this in a plugin rather than in the core class?

The methods are not even used in core so I'm wondering why the methods should even be defined in core. At this current moment, all the logic which we've added here are specific to the use case of the chat plugin so I see no reason for core to have to carry all the additional logic at this moment.

Reply:
The same could be said for anything in core that can be used for plugins, no? It doesn't make sense to me to have half of what the class does in core and half of it in a plugin...I'm sure we will find similar uses for it in core when we approach the existing issues around muting/ignoring users when sending personal messages. I don't think we should hold back on introducing useful code simply because it's not used in core yet.

Reply:
> The same could be said for anything in core that can be used for plugins, no?

Not really. I'm generally against adding things into core when it is not being used in core unless it is exposed through the proper plugin interfaces. Otherwise, we're just unnecessarily increasing the surface area of core. My personal opinion is that if we don't maintain strict rules about what to add to core, we would all just be adding generic but plugin specific logic into core. 

> I'm sure we will find similar uses for it in core when we approach the existing issues around muting/ignoring users when sending personal messages.

If we're sure about this then I think this is sufficient to justify its existence in core.

Reply:
I don't see it as plugin-specific at all. Just because it's being used by a plugin at this time does't mean it's plugin-specific. 

> If we're sure about this then I think this is sufficient to justify its existence in core.

Yes I am sure, that's why I have added it here ;) If I thought it would only ever be used for chat I would add it there.